pkg/scheduler/scheduler.go

```diff
@@ -400,6 +400,7 @@ func (e entryOrdering) Less(i, j int) bool {
 func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, w *workload.Info, message string) {
 	added := s.queues.RequeueWorkload(ctx, w)
```

Comment:
I think we might want to distinguish two scenarios of re-queueing (but fine to do in a follow up).

We have a case where we requeue because the workload didn't fit. But we also have a case where we requeue because we admitted workloads in another CQ that had higher priority, even though they might still fit in the next scheduling cycle. I don't think we should punish those workloads. They should skip the inadmissible stage.

Reply:
Agree it. 
As we just admit only one workload in the same Cohort in one scheduling cycle, maybe some other workloads are rejected directly and re-enqueue.  We need to distinguish these two scenarios and skip them.